README.md
![LogoKDM](https://raw.githubusercontent.com/ZR-TECDI/ZR_KDM/master/extras/kdm_logo.png)

[![Build Status](https://travis-ci.org/ZR-TECDI/ZR_KDM.svg?branch=master)](https://travis-ci.org/ZR-TECDI/ZR_KDM) ![GitHub release (latest by date including pre-releases)](https://img.shields.io/github/v/release/ZR-TECDI/ZR_KDM?include_prereleases) [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)

# ZR KDM o Kit de Desarrollo de Misiones ZR

Es una plantilla/framework que pueden utilizar editores de misiones para desarrollar misiones oficiales y extra-oficiales para nuestro querido clan. La intención es ayudar a los editores de misiones novatos y experimentados con una base sólida desde la cual comenzar su proyecto, además de entregarles varias funciones que alivian la carga de diseñar/programar acciones complejas y reducirlas a una sola línea de código.

## Uso

El KDM está diseñado para ser sencillo en su uso, tanto para editores experimentados como novatos. La mejor forma de comenzar es [descargar la última versión estable](https://github.com/ZR-TECDI/ZR_KDM/releases), que contiene una misión como plantilla en el mapa de Realidad Virtual. 

Si no sabes cómo utilizar esta plantilla en la misión que quieres hacer, utiliza ``zapador.bat`` de la carpeta raíz. **Zapador** es una herramienta que hace el trabajo inicial por ti, descarga la última versión posible, crea archivos de configuración básicos y mueve la misión a tu carpeta de misiones.

Instrucciones extendidas en el uso de la plantilla pueden encontrarse en nuestra documentación (PRÓXIMAMENTE)

## Autores

👤 **ZR-TECDI**

Los autores individuales de cada parte del proyecto están debidamente mencionados en la siguiente lista y en la documentación de su aporte particular.

<!-- Lista de colaboradores -->
<table style="height: 130px; margin-left: auto; margin-right: auto;" width="120">
<tbody>
<tr>
<td style="width: 110px;">
<p style="text-align: center;"><img src="https://github.com/corp-0.png?size=80" alt="" width="80" height="80" /></p>
<p style="text-align: center;"><a title="Enlace a github" href="https://github.com/corp-0" target="_blank" rel="noopener">Riquelme</a></p>
</td>
</tr>
</tbody>
</table>

## 🤝 Contribuir

Puedes contribuir con el proyecto informándonos de errores o haciendo sugerencias para mejorarlo, eres libre de comenzar un issue en [la página de issues](github.com/ZR-TECDI/ZR_KDM/issues). Por favor, sigue el formato sugerido.

Otra forma de contribuir es ponerse manos a la obra y ayudar en la fabricación del kit, escribiendo código y discutiendo sobre el mismo. Una guía completa sobre el entorno de trabajo de KDM y cómo escribir código con nosotros está disponible en la documentación (PRÓXIMAMENTE)


CHANGELOG.md
# [0.10.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.9.0...v0.10.0) (2020-07-12)


### Features

* **KDM:** Agregado script para recuperar loadout al respawnear ([de13aa2](https://github.com/ZR-TECDI/ZR_KDM/commit/de13aa22831ef7926f6857824694d9c48f58b59c))
* **KDM:** Agregado sistema de cajas de suministros ([2472f6a](https://github.com/ZR-TECDI/ZR_KDM/commit/2472f6a439aaf576259f7eaed1b0b73db81a9ca4))

# [0.9.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.8.0...v0.9.0) (2020-05-13)


### Bug Fixes

* **kdm:** arregla función para hacer diálogo en área ([2a4569f](https://github.com/ZR-TECDI/ZR_KDM/commit/2a4569f7efdeae45e9866f12612fb1a3f6c5fc9c))
* **kdm:** eliminada referencia a scripts eliminados previamente ([3ef7c08](https://github.com/ZR-TECDI/ZR_KDM/commit/3ef7c0872ae85516bcff68175e1e86d8ae0975f5))
* **kdm:** eliminado principal sospechoso de mem leak ([966498b](https://github.com/ZR-TECDI/ZR_KDM/commit/966498b905006662946cd78c9d043d2dfb15147c))
* **kdm:** plantilla actualizada correctamente ([7784cde](https://github.com/ZR-TECDI/ZR_KDM/commit/7784cdeab09f106e0e4bf1d7aa22142d39a27c39))
* **zapador:** corregido nombre de variable que causaba crash ([24c4fb9](https://github.com/ZR-TECDI/ZR_KDM/commit/24c4fb9b03c130502f832b79e621d069d61782d5))


### Features

* **kdm:** actualizada plantilla de personajes ([d9710e0](https://github.com/ZR-TECDI/ZR_KDM/commit/d9710e0ab34a993d65687ad00ef5e016142f4a9f))
* **kdm:** agregada función para diálogo npc privado ([557b707](https://github.com/ZR-TECDI/ZR_KDM/commit/557b70734971de66a73d0c3c7125d0380064d2ad))
* **kdm:** agregado script para izar bandera ([8d58675](https://github.com/ZR-TECDI/ZR_KDM/commit/8d58675c313b6c0fff43be6b3b7b2bd4cae02009))
* **kdm:** agregados toques de corneta a la plantilla ([09b5191](https://github.com/ZR-TECDI/ZR_KDM/commit/09b5191131b1c069d28f0ac1b2e37564851efb99))
* **kdm:** nuevas funciones para dialogo npc ([148b0fb](https://github.com/ZR-TECDI/ZR_KDM/commit/148b0fbd236d7627da8de39dc5385b537a948c4e))
* **zapador:** mejorado texto que comprueba versiones en consola ([ce04fb4](https://github.com/ZR-TECDI/ZR_KDM/commit/ce04fb4d37d61bad090c5a28f2663d735dda4c46))

# [0.8.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.7.0...v0.8.0) (2020-01-12)


### Features

* **kdm:** plantilla actualizada para nuevo ace medical ([1fe6880](https://github.com/ZR-TECDI/ZR_KDM/commit/1fe6880ea44bf9a0494026e584c4c6d7d4c0a126))
* **kdm):** plantilla actualizada para el último ace medical ([2046aad](https://github.com/ZR-TECDI/ZR_KDM/commit/2046aad57a62dd1f3e84250e8870906240176299))

# [0.7.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.6.1...v0.7.0) (2020-01-08)


### Bug Fixes

* **zapador:** corregida ruta de assets ([25e3d8a](https://github.com/ZR-TECDI/ZR_KDM/commit/25e3d8afc75a97ed49a84427c4bc7801a62c2619))
* **zapador:** corregido error al no tener config ([f77b28e](https://github.com/ZR-TECDI/ZR_KDM/commit/f77b28ebe60da963e867d08c980a5bcb78ff6d92))
* **zapador:** corregido problema de import distutils ([54515c8](https://github.com/ZR-TECDI/ZR_KDM/commit/54515c89ede4f8da32a081188963de8c24a2fe4a))
* **zapador:** incluir lib de glew ([e57da79](https://github.com/ZR-TECDI/ZR_KDM/commit/e57da79f6fea13f95336cdece9723db6eb4dd0d7))
* **zapador:** ya no se ejecuta manejador_plantilla dos veces la primera vez ([2f669ee](https://github.com/ZR-TECDI/ZR_KDM/commit/2f669eefd08a8fe62062e9736cbfaf8363ab3293))


### Features

* **template:**  .rar para distribuir con zapador ([e7766e1](https://github.com/ZR-TECDI/ZR_KDM/commit/e7766e1a3a8dbc0e7771f8192522d9671d30dd34))
* **zapador:** botón para abrir carpeta config ([ea7d074](https://github.com/ZR-TECDI/ZR_KDM/commit/ea7d07485f10b44f52dcd55a647b0631ce3eb0a7))
* **zapador:** importar y actualizar misiones hechas ([3798316](https://github.com/ZR-TECDI/ZR_KDM/commit/3798316a5dda7c7fe5a1475df030de795104a08f))
* **zapador:** primera versión de prueba pública ([9cfd9be](https://github.com/ZR-TECDI/ZR_KDM/commit/9cfd9beb5b444f28eb892b828454a0a6cf8a4f7c))
* **zapador:** zapador actualizado a v1.0.0 ([e83649c](https://github.com/ZR-TECDI/ZR_KDM/commit/e83649cef18a73f5587a560d8ac0161ef372d175))

## [0.6.1](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.6.0...v0.6.1) (2019-11-16)


### Bug Fixes

* **KDM:** corregido error que imprimía nombre de mision en lugar de tipo ([160406c](https://github.com/ZR-TECDI/ZR_KDM/commit/160406cb8a85e4342976c31c7dbc45aedc9586c3))

# [0.6.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.5.0...v0.6.0) (2019-11-15)


### Features

* **KDM:** Agregados espacios para ECHO ([5bdaf0c](https://github.com/ZR-TECDI/ZR_KDM/commit/5bdaf0cc915620b9df83943b3bdd4b9318620002))

# [0.5.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.4.0...v0.5.0) (2019-11-11)


### Bug Fixes

* **zapador:** corrección de dirección para auto-update ([3c5bcd9](https://github.com/ZR-TECDI/ZR_KDM/commit/3c5bcd978831faa262206c580d13d9bec19887ba))


### Features

* **zapador:** Nueva característica de auto update ([18a4673](https://github.com/ZR-TECDI/ZR_KDM/commit/18a46733e00a901ef9c927186e1601049f619e1a))

# [0.4.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.3.0...v0.4.0) (2019-11-08)


### Features

* **zapador:** Agregado Prei a la lista de mapas ([97e5d20](https://github.com/ZR-TECDI/ZR_KDM/commit/97e5d20c2dacd95586b4c66d60a99659e16cecd3))

# [0.3.0](https://github.com/ZR-TECDI/ZR_KDM/compare/v0.2.0...v0.3.0) (2019-10-08)


### Bug Fixes

* **kdm:** Eliminadas las malvadas tabs ([8e03405](https://github.com/ZR-TECDI/ZR_KDM/commit/8e03405))
* **zapador:** corregido error que hacía llamar dos veces la misma función ([1eadbf0](https://github.com/ZR-TECDI/ZR_KDM/commit/1eadbf0))


### Features

* **kdm:** Nueva función para imprimir datos de misión en el RPT ([85669fd](https://github.com/ZR-TECDI/ZR_KDM/commit/85669fd))
* **zapador:** agregada función para tomar eventos de conexión y desconexión de jugadores ([3922d65](https://github.com/ZR-TECDI/ZR_KDM/commit/3922d65))

herramientas/zapador/source/main.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
 
"""Zapador es un pequeño script que se encarga de la logística de actualizar el KDM a su última versión y
generar archivos para una nueva misión"""
 
# imports
import urllib.request
import requests
import os
import sys
import zipfile
import time
import shutil
from pathlib import Path
import generador_mision
import versionado
import json
from packaging import version
from subprocess import Popen, CREATE_NEW_CONSOLE

# globales
VERSION = versionado.version

primera_vez         = True
branch = 'master'
quiere_online        = False
dir_script          = os.path.dirname(os.path.realpath(sys.argv[0]))
directorio_descarga = dir_script + "\\descargas"

CONFIG_DICT = {
    'primera_vez': 'false',
    'mpmissions' : '',
    'online' : 'True',
    'branch' : 'master'
}


CONFIG_ONLINE       = ""
MPMISSIONS_DIR      = ""

main_menu = """
    +=======================+===============================================================================+
    |        Comando        |                                  Descripción                                  |
    +=======================+===============================================================================+
    | 1- Configurar KDM     | Cambia el comportamiento de KDM                                               |
    +-----------------------+-------------------------------------------------------------------------------+
    | 2- Crear misión       | Mueve KDM a tu carpeta de misiones y crea archivos base por ti                |
    +-----------------------+-------------------------------------------------------------------------------+
    | 3- Olvidar seteo      | Reinicia la configuración de Zapador                                          |
    +-----------------------+-------------------------------------------------------------------------------+
    | 4- Salir              | Cierra la aplicación                                                          |
    +-----------------------+-------------------------------------------------------------------------------+
"""
sepa = """   .     .     .     .     .     .     .     .     .     .
_.` `._.` `._.` `._.` `._.` `._.` `._.` `._.` `._.` `._.` `."""

AUTOR         = '"ZR Army - Dept. Tecnico y Diseño";'
NOMBRE_MISION = '"TEMPLATE ZR";'
quiere_foto   = False
FOTO_MISION   = '"img\loadscreen.jpg";'
DESC_MISION   = '"Uso exclusivo de Zona Roja Army.";'
EXT_MAPA      = '.VR'
 
# funciones

def descarga():
    """Descarga la última versión disponible del KDM en un directorio nuevo llamado 'descargas'"""

    global quiere_online

    url = "https://github.com/ZR-TECDI/ZR_KDM/archive/{}.zip".format(branch)

    if quiere_online:

        print(sepa)
        print("Descargando última versión de KDM desde: "+ url)

        try:
            os.mkdir(dir_script + "\\descargas")
        except Exception as e:
            print("ERROR AL CREAR DIRECTORIO DE DESCARGA: " + str(e))
            os.system("pause")
        else:
            print("Directorio de descarga temporal creado con éxito")
            os.chdir(dir_script + "\\descargas")

        try:
            ultimo_estable = urllib.request.urlretrieve(url, directorio_descarga+"\\kdm.zip")
        except Exception as e:
            print("ERROR AL DESCARGAR: "+ str(e))
            os.system("pause")
        else:
            ultimo_estable = ultimo_estable[0]
            print("Archivo descargado con éxito en: "+ultimo_estable)
            unzip(ultimo_estable)
    else:
        print(sepa)
        print('Modo ofline detectado, necesitarás poner una copia de KDM en una carpeta llamada "plantilla", junto a este ejecutable...')
        plantilla = dir_script + "\\plantilla\\template_ZR.VR"
        ruta_plantilla = Path(plantilla)
        if ruta_plantilla.exists():
            mover_carpeta(dir_script + '\plantilla')
        else:
            print('No encontramos una copia válida de KDM junto a este script.')
            print('Recuerda crear una carpeta y llamarla "plantilla" junto a el ejecutable de Zapador y poner la carpeta de KDM dentro.')
            os.system('pause')
            main()


def unzip(archivo):
    """Descomprime el archivo descargado"""

    print("Extrayendo archivo zip en " + directorio_descarga)
    with zipfile.ZipFile(archivo, 'r') as zip_ref:
        zip_ref.extractall(directorio_descarga)
    mover_carpeta(directorio_descarga)


def mover_carpeta(carpeta):
    """Mueve la carpeta de plantilla desde la descarga hasta la carpeta de misiones del editor"""

    global MPMISSIONS_DIR
    global quiere_online

    os.system("cls")
    os.chdir(carpeta)
    ruta_plantilla = Path(MPMISSIONS_DIR + "\\template_ZR.VR\\")
    if ruta_plantilla.exists():
        shutil.rmtree(MPMISSIONS_DIR + "\\template_ZR.VR")

    ruta = os.listdir(carpeta)

    for i in ruta:
        es_carpeta = os.path.isdir(i)
        if es_carpeta:
            try:
                if quiere_online:
                    shutil.move(i + "\\template_ZR.VR", MPMISSIONS_DIR)
                if not quiere_online:
                    shutil.move(i, MPMISSIONS_DIR)
            except Exception as e:
                print("ERROR AL MOVER PLANTILLA A CARPETA DE MISIÓN: ")
                print(str(e))
            else:
                print("Plantilla movida con éxito")
            finally:
                os.chdir(dir_script)
                shutil.rmtree(carpeta)
                os.system("pause")

    carpeta_final = MPMISSIONS_DIR + "\\template_ZR.VR"

    crear_archivos(carpeta_final)

def crear_archivos(carpeta_final):
    """Consulta al usuario una serie de datos para crear archivos de misión en base a una plantilla"""

    global AUTOR
    global NOMBRE_MISION
    global quiere_foto
    global FOTO_MISION
    global DESC_MISION
    global EXT_MAPA

    AUTOR         = input('Ingresa el autor de la misión (DEFAULT: "ZR Army - Dept. Tecnico y Diseño"):\n')
    if AUTOR == "":
        AUTOR = "ZR Army - Dept. Técnico y Diseño"

    NOMBRE_MISION = input('Ingresa el nombre de la misión (DEFAULT: "TEMPLATE ZR"):\n')
    if NOMBRE_MISION == "":
        NOMBRE_MISION = "TEMPLATE ZR"

    quiere_foto   = input('¿Deseas agregar una foto durante la carga de misión? [S/N]\n').lower()
    if quiere_foto == "s":
        FOTO_MISION = input('Ingresa la dirección relativa de la foto que quieres usar (DEFAULT: "img\loadscreen.jpg"):\n')
        if FOTO_MISION == "":
            FOTO_MISION = "img\loadscreen.jpg"
        quiere_foto = True
    else:
        quiere_foto = False

    DESC_MISION   = input('Ingresa una descripción breve para tu misión (DEFAULT: "Uso exclusivo de Zona Roja Army."):\n')
    if DESC_MISION == "":
        DESC_MISION = "Uso exclusivo de Zona Roja Army."

    print("Elige el mapa donde se va a realizar la misión:\n")
    print("1. Altis") #.Altis
    print("2. Stratis") #.Stratis
    print("3. Malden") #.Malden
    print("4. Realidad Virtual")#.VR
    print("5. Bozcaada")#.Bozcaada
    print("6. Bukovina")#.Bootcamp_ACR
    print("7. Bystrica")#.Woodland_ACR
    print("8. Chernarus")#.chernarus
    print("9. Chernarus (Verano)")#.chernarus_summer
    print("10. Chernarus (Invierno)")#.Chernarus_Winter
    print("11. Desierto")#.Desert_E
    print("12. Fallujah")#.fallujah
    print("13. G.O.S. Al Rayak")#.pja310
    print("14. G.O.S. Dariyah")#.pj307
    print("15. Hazar-Kot (valle)")#.MCN_HazarKot
    print("16. Kidal")#.Kidal
    print("17. Kunduz, Afganistán")#.Kunduz
    print("18. Porto")#.porto 
    print("19. Campo de pruebas")#.ProvingGrounds_PMC
    print("20. Rahmadi")#.intro 
    print("21. Sahrani")#.sara 
    print("22. Sahrani del Sur")#.saralite
    print("23. Sahrani Unida")#.sara_dbe1
    print("24. Shapur")#.Shapur_BAF
    print("25. Takistán")#.takistan
    print("26. Takistán (montañas)")#.Mountains_ACR
    print("27. Tanoa")#.Tanoa
    print("28. Utes")#.utes
    print("29. Zargabad")#.zargabad
    print('30. Prei (Camboya)')#.prei_khmaoch_luong

    EXT_MAPA = input("El número de tu elección:\n> ")

    if EXT_MAPA == '1':
        EXT_MAPA = '.Altis'
    elif EXT_MAPA == '2':
        EXT_MAPA = ".Stratis"
    elif EXT_MAPA == '3':
        EXT_MAPA = '.Malden'
    elif EXT_MAPA == '4':
        EXT_MAPA = '.VR'
    elif EXT_MAPA == '5':
        EXT_MAPA = '.Bozcaada'
    elif EXT_MAPA == '6':
        EXT_MAPA = '.Bootcamp_ACR'
    elif EXT_MAPA == '7':
        EXT_MAPA = '.Woodland_ACR'
    elif EXT_MAPA == '8':
        EXT_MAPA = '.chernarus'
    elif EXT_MAPA == '9':
        EXT_MAPA = '.chernarus_summer'
    elif EXT_MAPA == '10':
        EXT_MAPA = '.Chernarus_Winter'
    elif EXT_MAPA == '11':
        EXT_MAPA = '.Desert_E'
    elif EXT_MAPA == '12':
        EXT_MAPA = '.fallujah'
    elif EXT_MAPA == '13':
        EXT_MAPA = '.pja310'
    elif EXT_MAPA == '14':
        EXT_MAPA = '.pj307'
    elif EXT_MAPA == '15':
        EXT_MAPA = '.MCN_HazarKot'
    elif EXT_MAPA == '16':
        EXT_MAPA = '.Kidal'
    elif EXT_MAPA == '17':
        EXT_MAPA = '.Kunduz'
    elif EXT_MAPA == '18':
        EXT_MAPA = '.porto'
    elif EXT_MAPA == '19':
        EXT_MAPA = '.ProvingGrounds_PMC'
    elif EXT_MAPA == '20':
        EXT_MAPA = '.intro'
    elif EXT_MAPA == '21':
        EXT_MAPA = '.sara'
    elif EXT_MAPA == '22':
        EXT_MAPA = '.saralite'
    elif EXT_MAPA == '23':
        EXT_MAPA = '.sara_dbe1'
    elif EXT_MAPA == '24':
        EXT_MAPA = '.Shapur_BAF'
    elif EXT_MAPA == '25':
        EXT_MAPA = '.takistan'
    elif EXT_MAPA == '26':
        EXT_MAPA = '.Mountains_ACR'
    elif EXT_MAPA == '27':
        EXT_MAPA = '.Tanoa'
    elif EXT_MAPA == '28':
        EXT_MAPA = '.utes'
    elif EXT_MAPA == '29':
        EXT_MAPA = '.zargabad'
    elif EXT_MAPA == '30':
        EXT_MAPA = '.prei_khmaoch_luong'
    else:
        print('Error, la opción elegida no es válida. Usando valor DEFAULT (Realidad virtual)')
        EXT_MAPA = '.VR'
        os.system('pause')

    os.system('cls')
    generador_mision.main(carpeta_final, AUTOR, NOMBRE_MISION, quiere_foto, FOTO_MISION, DESC_MISION, EXT_MAPA, MPMISSIONS_DIR)


def set_configuracion():
    """Crea o sobrescribe el archivo desde el que se toma la ruta a la carpeta de misiones"""

    global MPMISSIONS_DIR
    global CONFIG_ONLINE
    global CONFIG_DICT
    global branch

    os.system("cls")
    with open(dir_script + '\zapador_config.json', 'r') as fp:
        CONFIG_DICT = json.load(fp)

    MPMISSIONS_DIR = CONFIG_DICT['mpmissions']
    CONFIG_ONLINE = CONFIG_DICT['online']

    print("1. Carpeta de misión : {}".format(MPMISSIONS_DIR))
    print("2. Descargar siempre última versión : {}".format(CONFIG_ONLINE))
    print("3. Versión: {}".format(branch))
    print("0. Volver al menú principal")
    print()
    print("¿Qué configuración quieres cambiar?")
    choice = input("> ")

    if choice == "0":
        leer_configuracion()
        return 0
    elif choice == "1":
        value = input("Ingrese nuevo valor para esta configuración: \n>")
        value = value.lower().capitalize()
        CONFIG_DICT['mpmissions'] = value
        with open (dir_script + '\zapador_config.json', 'w') as fp:
            json.dump(CONFIG_DICT, fp)
    elif choice == "2":
        value = input("Ingrese nuevo valor para esta configuración: \n>")
        if value.lower() not in ["true", "false"]:
            print('Valor ingresado no es válido para esta configuración. Opciones: True/False')
            os.system('pause')
            set_configuracion()
        else:
            CONFIG_DICT['online'] = value.lower(). capitalize()
            with open (dir_script + '\zapador_config.json', 'w') as fp:
                json.dump(CONFIG_DICT, fp)
    elif choice == "3":
        value = input("Ingresa qué branch quieres usar (experto): \n>")
        if value.lower() not in ["master", "dev-branch"]:
            print('Valor ingresado no es válido para esta configuración. Opciones: master/dev-branch\n')
            os.system('pause')
            set_configuracion()
        else:
            branch = value.lower()
            CONFIG_DICT['branch'] = value.lower()
            with open (dir_script + '\zapador_config.json', 'w') as fp:
                json.dump(CONFIG_DICT, fp)

    os.system('cls')
    print("Configuración actualizada...")
    os.system('pause')
    set_configuracion()


def olvidar_configuracion():
    """Elimina el archivo desde donde se toma la carpeta de misiones"""

    os.system("cls")
    try:
        os.remove(dir_script + "/zapador_config.json")
    except Exception as e:
        print("ERROR AL INTENTAR ELIMINAR CONFIGURACIÓN: ")
        print(str(e))
    else:
        print("Configuración eliminada con éxito")
    finally:
        os.system("pause")
    
    leer_configuracion()

def intro():
    """Presenta la aplicación con su versión y créditos"""

    os.system('cls')
    print()
    print("""
                       _____________________________________________________
                      |                                                     |
            / _____ | |                       ZAPADOR                       |
           / /(__) || |                       v{}                        |
             _______  |                                                     |
  ________/ / |OO| || |              Escrito por: Riquelme                  |
 |         |-------|| |                                                     |
(|         |     -.|| |_______________________                              |
 |  ____   \       ||_________||____________  |             ____      ____  |
/| / __ \   |______||     / __ \   / __ \   | |            / __ \    / __ \ |
\|| /  \ |_______________| /  \ |_| /  \ |__| |___________| /  \ |__| /  \|_|/
   | () |                 | () |   | () |                  | () |    | () |
    \__/                   \__/     \__/                    \__/      \__/

    """.format(VERSION))
    print()
    os.system('pause')
    main()

def leer_configuracion():

    global primera_vez
    global quiere_online
    global branch
    global CONFIG_DICT
    global MPMISSIONS_DIR
    global CONFIG_ONLINE

    try: 
        with open(dir_script + "/zapador_config.json", 'r') as f:
            CONFIG_DICT = json.load(f)
            primera_vez = CONFIG_DICT['primera_vez']
    except:
        primera_vez = True
    else:
        primera_vez = False

    if primera_vez:
        os.system('cls')
        print("Bienvenido a Zapador.\n")
        print("Este programa fue diseñado con la intención de facilitar el pie inicial a la creación de misiones.")
        print("A continuación, necesitamos que nos des algo de información para configurar todo.\n")
        print()
        print("Busca la ruta a tu carpeta de misiones, debería verse algo así: ")
        print("C:\\Users\TU-USUARIO\Documents\Arma 3 - Other Profiles\TU-PERFIL-DE-ARMA\mpmissions")
        MPMISSIONS_DIR = input("Ingresa la ruta ahora: ")
        CONFIG_DICT['mpmissions'] = MPMISSIONS_DIR
        primera_vez = False
        data = json.dumps(CONFIG_DICT)

        with open (dir_script+'\zapador_config.json', 'w')as f:
            f.write(data)
            CONFIG_DICT = json.loads(data)
        leer_configuracion()

    else:
        CONFIG_ONLINE = CONFIG_DICT['online']
        branch = CONFIG_DICT['branch']
        MPMISSIONS_DIR = CONFIG_DICT['mpmissions']

        if CONFIG_ONLINE == 'True':
            quiere_online = True
            auto_update()
        else:
            quiere_online = False
            intro()

def auto_update():
    """Compara la versión del programa con la última publicada y actualiza si es menor"""
    global VERSION
    global branch

    try:
        url = "https://raw.githubusercontent.com/ZR-TECDI/ZR_KDM/{}/herramientas/zapador/versionado.py".format(branch)
        req = requests.get(url)
        estable = req.text.replace('"','')
        estable = estable.replace('version=','')


        if req.ok:
            if version.parse(VERSION) >= version.parse(estable):
                intro()
            else:
                print('Hay una nueva versión en línea. Actualizando...')
                print('Última versión estable: ' + estable + ' || Versión en uso: '+ VERSION)
                os.system('pause')
                Popen(dir_script + '/auto_update.exe', creationflags=CREATE_NEW_CONSOLE)
                quit()
                
        else:
            intro()
    except:
        pass

def main():
    """Menú principal de la aplicación"""

    global quiere_online

    loop = True
    while loop:
        os.system("cls")
        print(main_menu)

        choice = input("> ")

        if choice == "1":
            set_configuracion()
        if choice == "2":
            descarga()
        if choice == "3":
            olvidar_configuracion()
        if choice == "4":
            loop = False
            sys.exit()

if __name__ == "__main__":
	leer_configuracion()